sunpy/coordinates/transformations.py

```diff
@@ -103,6 +103,11 @@ def hcc_to_hpc(helioccoord, heliopframe):
     """
     Convert from Heliocentic Cartesian to Helioprojective Cartesian.
     """
+    if helioccoord.observer != heliopframe.observer:
+        raise ConvertError("Cannot transform heliocentric coordinates to "
```

Comment:
maybe "Cannot transform directly..." since you actually can do the transformation, you just have to go through HGS first.